src/main/java/com/elina/railwayApp/controller/UserController.java

```diff
@@ -28,48 +27,27 @@
     @Autowired
     private UserService userService;
 
-    @Autowired
-    private RoleService roleService;
-
-    @Autowired
-    private MailService mailService;
 
-    @RequestMapping(value = URLs.REGISTRATION, method = RequestMethod.POST)
-    public ResponseEntity<?> registration(@RequestBody User user) throws IOException {
-        log.info("PREPARING PROCESS FOR REGISTER");
-        if (userService.findByEmail(user.getLogin()) == null && user != null) {
-            Role role = roleService.getRole();
-            Set<Role> roleSet = new HashSet<>();
-            roleSet.add(role);
-            user.setPassword(Utils.encodePassword(user.getPassword()));
-            user.setRoles(roleSet);
-            userService.add(user);
-            log.info("REGISTRATION WAS SUCCESSFUL FOR USER login = " + user.getLogin());
-            Message message = Message.createWelcomeMessage(user.getLogin());
-            mailService.sendMimeMessage(message);
-            return ResponseEntity.ok("registration success");
+    @PostMapping(URLs.REGISTRATION)
+    public ResponseEntity<?> registration(@RequestBody User user) {
+        try {
+            userService.registration(user);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (UserNotFoundException userNotFoundException) {
+            userNotFoundException.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        } catch (Exception messagingException) {
```

Comment:
Catching of exception is a bad practice.
1. Use Spring error handler.
2. Define specific exception if user not found in DAO